### PR TITLE
Update abinit version

### PR DESCRIPTION
**Pull request by downloadico**

### CODE REVIEW

**Review by tldahlgren — COMMENTED** [review]
Confirmed the sha256.

File: var/spack/repos/builtin/packages/abinit/package.py [hunk 1]
```diff
@@ -157,6 +161,10 @@ def configure_args(self):
                     ]
                 )
         else:
+            if "@:9.8" in spec:
+                oapp("--with-fftw {0}".format(spec["fftw"].prefix))
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/abinit/package.py]: I am unable to build this package:

```
...
Error: KeyError: 'No spec with name fftw in abinit@9.8.3%gcc@8.3.1~install-tests~libxml2+mpi~openmp~scalapack~wannier90 build_system=autotools optimization-flavor=standard patches=6c7e501 arch=linux-rhel7-broadwell ^atompaw@4.2.0.1%gcc@8.3.1 build_system=autotools arch=linux-rhel7-broadwell ^autoconf@2.69%gcc@8.3.1 build_system=autotools patches=7793209 arch=linux-rhel7-broadwell ^automake@1.13.4%gcc@8.3.1 build_system=autotools arch=linux-rhel7-broadwell ^ca-certificates-mozilla@2022-10-11%gcc@8.3.1 build_system=generic arch=linux-rhel7-broadwell ^cmake@3.26.0%gcc@8.3.1~doc+ncurses+ownlibs~qt build_system=generic build_type=Release arch=linux-rhel7-broadwell ^cpio@2.11%gcc@8.3.1 build_system=autotools arch=linux-rhel7-broadwell ^hdf5@1.8.22%gcc@8.3.1~cxx~fortran+hl~ipo+mpi+shared~szip~threadsafe+tools api=default build_system=cmake build_type=RelWithDebInfo arch=linux-rhel7-broadwell ^hwloc@2.9.0%gcc@8.3.1~cairo~cuda~gl~libudev+libxml2~netloc~nvml~oneapi-level-zero~opencl+pci~rocm build_system=autotools libs=shared,static arch=linux-rhel7-broadwell ^intel-mkl@2020.4.304%gcc@8.3.1~ilp64+shared build_system=generic threads=none arch=linux-rhel7-broadwell ^libevent@2.1.12%gcc@8.3.1+openssl build_system=autotools arch=linux-rhel7-broadwell ^libiconv@1.16%gcc@8.3.1 build_system=autotools libs=shared,static arch=linux-rhel7-broadwell ^libpciaccess@0.16%gcc@8.3.1 build_system=autotools arch=linux-rhel7-broadwell ^libtool@2.4.7%gcc@8.3.1 build_system=autotools arch=linux-rhel7-broadwell ^libxc@5.2.3%gcc@8.3.1~cuda+shared build_system=autotools arch=linux-rhel7-broadwell ^libxml2@2.10.1%gcc@8.3.1~python build_system=autotools arch=linux-rhel7-broadwell ^m4@1.4.16%gcc@8.3.1+sigsegv build_system=autotools arch=linux-rhel7-broadwell ^ncurses@5.9.20130511%gcc@8.3.1~symlinks+termlib abi=none build_system=autotools patches=f84b270 arch=linux-rhel7-broadwell ^netcdf-c@4.9.0%gcc@8.3.1~dap~fsync~hdf4~jna+mpi+optimize~parallel-netcdf+pic+shared+zstd build_system=autotools arch=linux-rhel7-broadwell ^netcdf-fortran@4.6.0%gcc@8.3.1~doc+pic+shared build_system=autotools arch=linux-rhel7-broadwell ^numactl@2.0.14%gcc@8.3.1 build_system=autotools patches=4e1d78c,62fc8a8,ff37630 arch=linux-rhel7-broadwell ^openmpi@4.1.4%gcc@8.3.1~atomics~cuda~cxx~cxx_exceptions~gpfs~internal-hwloc~java~legacylaunchers~lustre~memchecker~orterunprefix+romio+rsh~singularity+static+vt+wrapper-rpath build_system=autotools fabrics=none schedulers=none arch=linux-rhel7-broadwell ^openssh@7.4p1%gcc@8.3.1+gssapi build_system=autotools arch=linux-rhel7-broadwell ^openssl@3.0.7%gcc@8.3.1~docs~shared build_system=generic certs=mozilla arch=linux-rhel7-broadwell ^perl@5.16.3%gcc@8.3.1+cpanm+shared+threads build_system=generic patches=0eac10e,3bbd7d6 arch=linux-rhel7-broadwell ^pkg-config@0.27.1%gcc@8.3.1+internal_glib build_system=autotools patches=49ffcd6 arch=linux-rhel7-broadwell ^pkgconf@1.8.0%gcc@8.3.1 build_system=autotools arch=linux-rhel7-broadwell ^pmix@4.1.2%gcc@8.3.1~docs+pmi_backwards_compatibility~python~restful build_system=autotools arch=linux-rhel7-broadwell ^util-macros@1.19.3%gcc@8.3.1 build_system=autotools arch=linux-rhel7-broadwell ^xz@5.2.2%gcc@8.3.1~pic build_system=autotools libs=shared,static arch=linux-rhel7-broadwell ^zlib@1.2.13%gcc@8.3.1+optimize+pic+shared build_system=makefile arch=linux-rhel7-broadwell ^zstd@1.5.2%gcc@8.3.1~programs build_system=makefile libs=shared,static arch=linux-rhel7-broadwell'
...
```

**Review by tldahlgren — CHANGES_REQUESTED** [review]
I am unable to build the package due to an issue with the `fftw` dependency identification.

File: var/spack/repos/builtin/packages/abinit/package.py (hunk 1 above)
Comment (downloadico) [reply var/spack/repos/builtin/packages/abinit/package.py]: Thanks for catching that!  I should have been looking for fftw-api, not fftw.

File: var/spack/repos/builtin/packages/abinit/package.py [hunk 2]
```diff
@@ -157,6 +161,10 @@ def configure_args(self):
                     ]
                 )
         else:
+            if "@:9.8" in spec:
+                oapp("--with-fftw {0}".format(spec["fftw-api"].prefix))
+                oapp("--with-hdf5 {0}".format(spec["hdf5"].prefix))
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/abinit/package.py]: You also need the following to get past `configure`:

```suggestion
                oapp("--with-fftw={0}".format(spec["fftw-api"].prefix))
                oapp("--with-hdf5={0}".format(spec["hdf5"].prefix))
```

Comment (tldahlgren) [reply var/spack/repos/builtin/packages/abinit/package.py]: I *can* build successfully with the suggested change above.

**Review by tldahlgren — APPROVED** [review]

**Pull request merged**